flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -159,42 +217,65 @@ public String getShardIterator(KinesisStreamShard shard, String shardIteratorTyp
+	 * <a href="http://google.com">https://www.awsarchitectureblog.com/2015/03/backoff.html</a>. This is necessary
+	 * because concurrent calls will be made by all parallel subtask's {@link ShardDiscoverer}s. This jitter backoff
+	 * approach will help distribute calls across the discoverers over time.
+	 *
 	 * @param streamName the stream to describe
 	 * @param startShardId which shard to start with for this describe operation (earlier shard's infos will not appear in result)
 	 * @return the result of the describe stream operation
 	 */
-	private DescribeStreamResult describeStream(String streamName, String startShardId) {
+	private DescribeStreamResult describeStream(String streamName, String startShardId) throws InterruptedException {
 		final DescribeStreamRequest describeStreamRequest = new DescribeStreamRequest();
 		describeStreamRequest.setStreamName(streamName);
 		describeStreamRequest.setExclusiveStartShardId(startShardId);
 
 		DescribeStreamResult describeStreamResult = null;
 		String streamStatus = null;
-		int remainingRetryTimes = Integer.valueOf(
-			configProps.getProperty(KinesisConfigConstants.CONFIG_STREAM_DESCRIBE_RETRIES, Integer.toString(KinesisConfigConstants.DEFAULT_STREAM_DESCRIBE_RETRY_TIMES)));
-		long describeStreamBackoffTimeInMillis = Long.valueOf(
-			configProps.getProperty(KinesisConfigConstants.CONFIG_STREAM_DESCRIBE_BACKOFF, Long.toString(KinesisConfigConstants.DEFAULT_STREAM_DESCRIBE_BACKOFF)));
 
-		// Call DescribeStream, with backoff and retries (if we get LimitExceededException).
-		while ((remainingRetryTimes >= 0) && (describeStreamResult == null)) {
+		// Call DescribeStream, with full-jitter backoff (if we get LimitExceededException).
+		Random seed = null;
+		int attemptCount = 0;
+		while (describeStreamResult == null) { // retry until we get a result
 			try {
 				describeStreamResult = kinesisClient.describeStream(describeStreamRequest);
 				streamStatus = describeStreamResult.getStreamDescription().getStreamStatus();
 			} catch (LimitExceededException le) {
-				LOG.warn("Got LimitExceededException when describing stream " + streamName + ". Backing off for "
-					+ describeStreamBackoffTimeInMillis + " millis.");
-				try {
-					Thread.sleep(describeStreamBackoffTimeInMillis);
-				} catch (InterruptedException ie) {
-					LOG.debug("Stream " + streamName + " : Sleep  was interrupted ", ie);
+				if (seed == null) {
+					seed = new Random();
 				}
+				long backoffMillis = fullJitterBackoff(
+					describeStreamBaseBackoffMillis, describeStreamMaxBackoffMillis, describeStreamExpConstant, attemptCount++, seed);
+				LOG.warn("Got LimitExceededException when describing stream " + streamName + ". Backing off for "
+					+ backoffMillis + " millis.");
+				Thread.sleep(backoffMillis);
 			} catch (ResourceNotFoundException re) {
 				throw new RuntimeException("Error while getting stream details", re);
 			}
-			remainingRetryTimes--;
 		}
 
 		if (streamStatus == null) {
```

Comment:
The `RuntimeException` below has a typo.
`Can't get stream info from ____ after`.

Also, I wonder where the number 3 is coming from.